app/models/spree/line_item_decorator.rb

```diff
@@ -20,7 +20,7 @@
 
   before_destroy :update_inventory_before_destroy
 
-  delegate :unit_description, to: :variant
+  delegate :product, :unit_description, to: :variant
```

Comment:
why is this needed exactly?

Reply:
Because on the original LineItem model, `#product` is defined as another association using `through: :variant`, and we've basically removed that association.